### PR TITLE
Data Base connection

### PR DESCRIPTION
**Pull request by Cxristian**
implements query and connections

### CODE REVIEW

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package com.jala.data;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * This class defines the connection and creation of the database
+ * @author by Cristian Lujan
+ * @Version 21/02/2019/C
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/ConnectionDB.java]: the version should be number and not date, please review the previous files merged to develop

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/ConnectionDB.java]: remove extra line

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    private void initialize() throws SQLException, ClassNotFoundException {
+        Class.forName("org.sqlite.JDBC");
+        connection = DriverManager.getConnection("jdbc:sqlite:dataSearch.db");
+        Statement state = connection.createStatement();
+        state.execute("CREATE TABLE if not exists criteriaSearch (id integer,criteria varchar(200), primary key(id));");
+    }
+
+    /**
+     * @return The connection
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/ConnectionDB.java:73]: add documentation for this method

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.PreparedStatement;
+
+/**
+ * This class defines the Querys Generals for the database
+ * @author by Cristian Lujan
+ * @Version 21/02/2019/C
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: use version number

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ */
+
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: remove extra line

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+     * Constructor to initialize the connection
+     * @throws SQLException
+     * @throws ClassNotFoundException
+     */
+    public QueryGeneral() throws SQLException, ClassNotFoundException {
+        connection = ConnectionDB.getInstance().getConnection();
+    }
+
+    /**
+     * Method
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: we need to improve this documentation

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+     * @throws SQLException
+     */
+    public ResultSet displayData() throws SQLException {
+        Statement state = connection.createStatement();
+        ResultSet resSet = ((Statement) state).executeQuery("SELECT id, criteria fileName FROM criteriaSearch");
+        return resSet;
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: Add documentation

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+     */
+    public void addCriteria(String criteriaJson) throws SQLException {
+        PreparedStatement prepared = connection.prepareStatement("INSERT INTO criteriaSearch(id, criteria) values(?,?);");
+        prepared.setInt(1, 1);
+        prepared.setString(2, criteriaJson);
+        prepared.execute();
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: Add documentation

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+     * @throws SQLException
+     */
+    public void updateCriteria(String criteriaJson) throws SQLException {
+        PreparedStatement prepared = connection.prepareStatement("UPDATE criteriaSearch SET criteria = ? ");
+        prepared.setString(2, criteriaJson);
+        prepared.executeUpdate();
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java]: Add documentation

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+     * @param idCriteria
+     * @throws SQLException
+     */
+    public void deleteCriteria(int idCriteria) throws SQLException {
+        PreparedStatement prepared = connection.prepareStatement("DELETE FROM criteriaSearch WHERE id = ?;");
+        prepared.setInt(1,idCriteria);
+        prepared.execute();
+    }
+}
```
Comment (jpsandovaln) [inline src/main/java/com/jala/data/QueryGeneral.java:75]: add extra line at end of file

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * This class defines the connection and creation of the database
+ * @author by Cristian Lujan
+ * @Version 21/02/2019/C
+ */
+
+public class ConnectionDB {
+
+    private static Connection connection;
+    private static ConnectionDB connectDB;
```
Comment (AKatrin) [inline src/main/java/com/jala/data/ConnectionDB.java:35]: Add documentation

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        Class.forName("org.sqlite.JDBC");
+        connection = DriverManager.getConnection("jdbc:sqlite:dataSearch.db");
+        Statement state = connection.createStatement();
+        state.execute("CREATE TABLE if not exists criteriaSearch (id integer,criteria varchar(200), primary key(id));");
+    }
+
+    /**
+     * @return The connection
+     */
+    public Connection getConnection(){
```
Comment (AKatrin) [inline src/main/java/com/jala/data/ConnectionDB.java]: Add a space before {

**Review by AKatrin — CHANGES_REQUESTED** [review]

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+import java.sql.Statement;
+
+/**
+ * This class defines the connection and creation of the database
+ * @author by Cristian Lujan
+ * @Version 0.0.1
+ */
+public class ConnectionDB {
+
+    private static Connection connection;
```
Comment (AKatrin) [inline src/main/java/com/jala/data/ConnectionDB.java:30]: add documentation

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    private static ConnectionDB connectDB;
```
Comment (AKatrin) [inline src/main/java/com/jala/data/ConnectionDB.java:35]: add documentation

File: src/main/java/com/jala/data/ConnectionDB.java [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+     * @throws SQLException
+     * @throws ClassNotFoundException
+     */
+    private ConnectionDB() throws SQLException, ClassNotFoundException {
+        initialize();
+    }
+
+    /**
+     * Method that returns the connection to the database
+     * @return Connectionqwe of Data Base
```
Comment (AKatrin) [inline src/main/java/com/jala/data/ConnectionDB.java]: Connectionqwe???

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+import java.sql.PreparedStatement;
+
+/**
+ * This class defines the Querys Generals for the database
+ * @author by Cristian Lujan
+ * @Version 0.0.1
+ */
+public class QueryGeneral {
+
+    private Connection connection;
```
Comment (AKatrin) [inline src/main/java/com/jala/data/QueryGeneral.java:31]: add documentation please

File: src/main/java/com/jala/data/QueryGeneral.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+     * @param idCriteria
+     * @throws SQLException
+     */
+    public void deleteCriteria(int idCriteria) throws SQLException {
+        PreparedStatement prepared = connection.prepareStatement("DELETE FROM criteriaSearch WHERE id = ?;");
+        prepared.setInt(1,idCriteria);
+        prepared.execute();
+    }
+    //TODO implement the update method
+}
```
Comment (AKatrin) [inline src/main/java/com/jala/data/QueryGeneral.java:75]: you do a pull of develop branch and do merge

**Review by AKatrin — CHANGES_REQUESTED** [review]

**Pull request merged**